hw/top_earlgrey/data/clocks.xdc

```diff
@@ -72,39 +90,166 @@ set_input_delay  -add_delay -clock_fall -clock jtag_tck -min  0.0 [get_ports {IO
+
+## For half-cycle
+#set_output_delay -clock clk_spi -min ${spi_dev_out_hold}  ${spi_dev_data} -add_delay
+#set_output_delay -clock clk_spi -max ${spi_dev_out_setup} ${spi_dev_data} -add_delay
+
+## For full-cycle
+set_output_delay -clock clk_spi -clock_fall -min ${spi_dev_out_hold}  ${spi_dev_data} -add_delay
+set_output_delay -clock clk_spi -clock_fall -max ${spi_dev_out_setup} ${spi_dev_data} -add_delay
+
+## set clock sense on the input to spi buffers to help the tool understand the
+## clocks are shifted versions of each other
+set_clock_sense -positive \
+  [get_pins -filter {DIRECTION == OUT && IS_LEAF} -of_objects \
+    [get_nets -segments -of_objects \
+      [get_pins top_earlgrey/u_spi_device/u_passthrough/u_pt_sck_cg/gen_xilinx.u_impl_xilinx/gen_gate.gen_bufgce.u_bufgce/I0] \
+    ] \
+  ] \
+  -clocks clk_spi
+
+set_clock_sense -negative \
+  [get_pins -filter {DIRECTION == OUT && IS_LEAF} -of_objects \
+    [get_nets -segments -of_objects \
+      [get_pins top_earlgrey/u_spi_device/u_clk_spi_out_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufr.bufr_i/I] \
+    ] \
+  ] \
+  -clocks clk_spi
+
+set_clock_sense -positive \
+  [get_pins -filter {DIRECTION == OUT && IS_LEAF} -of_objects \
+    [get_nets -segments -of_objects \
+      [get_pins top_earlgrey/u_spi_device/u_clk_spi_in_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufr.bufr_i/I] \
+    ] \
+  ] \
+  -clocks clk_spi
+
 create_generated_clock -name clk_spi_in  -divide_by 1 -source [get_ports SPI_DEV_CLK] [get_pins top_*/u_spi_device/u_clk_spi_in_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufr.bufr_i/O]
 create_generated_clock -name clk_spi_out -divide_by 1 -source [get_ports SPI_DEV_CLK] [get_pins top_*/u_spi_device/u_clk_spi_out_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufr.bufr_i/O] -invert
 
+## SPI Passthrough constraints
+create_generated_clock -name clk_spi_pt -divide_by 1 -source [get_ports SPI_DEV_CLK] [get_ports SPI_HOST_CLK]
+
+set spi_pt_data [get_ports {SPI_HOST_D0 SPI_HOST_D1 SPI_HOST_D2 SPI_HOST_D3}]
+set spi_host_board_skew 0.5
+set spi_host_board_delay 0.6
+set spi_host_out_hold      [expr -3.0 - ${spi_host_board_skew}]
+set spi_host_out_setup     [expr  3.0 + ${spi_host_board_skew}]
+set spi_host_in_delay_min 0
+set spi_host_in_delay_max [expr  9.0 + 2 * ${spi_host_board_delay}]
+
+set_output_delay -clock clk_spi_pt -min ${spi_host_out_hold}  ${spi_pt_data} -add_delay
+set_output_delay -clock clk_spi_pt -max ${spi_host_out_setup} ${spi_pt_data} -add_delay
+set_output_delay -clock clk_spi_pt -min ${spi_host_out_hold}  [get_ports SPI_HOST_CS_L] -add_delay
+set_output_delay -clock clk_spi_pt -max ${spi_host_out_setup} [get_ports SPI_HOST_CS_L] -add_delay
+
+set_input_delay  -clock clk_spi_pt -clock_fall -min ${spi_host_in_delay_min} \
+    ${spi_pt_data} -add_delay
+set_input_delay  -clock clk_spi_pt -clock_fall -max ${spi_host_in_delay_max} \
+    ${spi_pt_data} -add_delay
```

Comment:
Is this also covering the input paths for regular SPI host mode?

Reply:
No, those are below, in the "SPI Host constraints" section. The clocks are different, and we need I/O delays for each clock used.

Reply:
Ah yes you are right. I misread that part.